tests/utils/test_metadata_routing.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+import pytest
+from sklearn import config_context
+from sklearn.base import BaseEstimator, ClassifierMixin, MetaEstimatorMixin, clone
+
+from sklearn_compat._sklearn_compat import parse_version, sklearn_version
+from sklearn_compat.utils.metadata_routing import _raise_for_params
+
+
+@pytest.mark.skipif(
+    sklearn_version < parse_version("1.3"),
+    reason="Metadata routing was introduced in scikit-learn 1.3",
+)
+def test_process_routing():
+    """This test is used to check that what we mentioned in the documentation is
+    correct.
+    """
+    from sklearn.utils.metadata_routing import (
+        MetadataRouter,
+        MethodMapping,
+        process_routing,
+    )
+
+    class ExampleClassifier(ClassifierMixin, BaseEstimator):
+        def fit(self, X, y, sample_weight=None):
+            self.sample_weight_ = sample_weight
+            # all classifiers need to expose a classes_ attribute once they're fit.
+            self.classes_ = np.array([0, 1])
+            return self
+
+    class MetaClassifier(MetaEstimatorMixin, ClassifierMixin, BaseEstimator):
+        def __init__(self, estimator):
+            self.estimator = estimator
+
+        def fit(self, X, y, sample_weight=None):
+            routed_params = process_routing(self, "fit", sample_weight=sample_weight)
```

Comment:
@adrinjalali it seems that the CI shows that you cannot have the same call to the function in sklearn 1.3 and sklearn 1.4+, isn't it?

You would need to check the signature to know if you should `other_params` by position thus it comes back to what we were doing when importing.

Reply:
Yeah this is the signature

https://github.com/scikit-learn/scikit-learn/blob/a3100bcc8fbd8659ab5e7c80339b6ad53b623eb9/sklearn/utils/_metadata_requests.py#L1264

It needs `other_params`, so yeah, we need to pass `None` to it here.